### ISSUE TITLE
Way to unset `X509_VERIFY_PARAM` fields

### ISSUE THREAD

**Issue by daurnimator** [openssl/openssl#2673]
Currently there is no API to clear various members of a `X509_VERIFY_PARAM` e.g. an IP set with `X509_VERIFY_PARAM_set1_ip`

Potential solutions:

  - A flag (modifiable via `X509_VERIFY_PARAM_set_flags`) that allows to opt-out of a field
  - `X509_VERIFY_PARAM_set1_ip(param, NULL, 0)` should clear the value

**Comment by daurnimator** [comment]
Anyone able to look at this? With the structure being opaque in 1.1.0 I can't hack around it at the project level.

**Comment by daurnimator** [comment]
Ping?

**Comment by richsalz** [comment]
Yes it's a bug caused by the opaque structure work.  Someone will get around to it, or you can create a PR.

**Comment by daurnimator** [comment]
I can send a PR. Any preferences on the two solutions I mentioned in OP?

**Comment by mattcaswell** [comment]
The second option looks good to me.

**Comment by mattcaswell** [comment]
Adding new getters still seems like a reasonable thing to want to add. So if that resolves your issue - then go for it (although perhaps it should be `X509_VERIFY_PARAM_get0_ip` or a `get1` equivalent)?

**Comment by mattcaswell** [comment]
This impacts 1.1.0 too, so removing the 1.1.1 milestone.

**Comment by levitte** [comment]
Hmmm, is this we actually intend to fix, or are we waiting for a PR from @daurnimator?  This very much looks like everyone is waiting for someone else to actually code this while not saying it out loud.

**Comment by daurnimator** [comment]
I've just opened up my editor now. Hopefully will send a PR in an hour or so.

**Comment by daurnimator** [comment]
Let me know if the style in #6932 is okay; I'd like to add the same functionality for host (and I guess email for completeness).

Also, I need to sign the CLA: what is the fax number? (is there someone available to take this private?)

**Comment by daurnimator** [comment]
Ping?

**Comment by t8m** [comment]
Is this still valid?

**Comment by t8m** [comment]
The version 1.1.1 is not supported anymore. Closing. If this is still an issue with currently supported releases, please open a new issue.
